intermediate-exercises/src/components/02-StopWatch.js

```diff
@@ -1,26 +1,38 @@
 import React, { Component } from 'react';
 
-/*
-* Exercise 2:
-*
-*  Create a `StopWatch` component that has a Start/Stop button and a Clear
-*  button.
-*
-*  Pressing Start will start a timer and the lapsed time in
-*  milliseconds should be displayed above the buttons.
-*
-*  Once started the Start button should change to Stop. Clicking Stop
-*  will stop the timer but lapsed time will be preserved.
-*
-*  Clicking Start again will resume the timer from where it left off.
-*
-*  Clicking Clear will stop the timer if it's running and reset the lapsed time to 0.
-*/
-
 class StopWatch extends Component {
+  state = {
+    stopWatchStart: false,
+    elapsedTime: 0
+  };
+
+  toggleTimer = () => {
+    this.setState({ stopWatchStart: this.state.stopWatchStart === false });
+  };
+
+  startTimer = () => {
+    setTimeout(() => {
+      this.setState({ elapsedTime: this.state.elapsedTime + 1 });
```

Comment:
Maybe a good idea is to use the other version of `setState` (the one using the function) since we are using the state to generate the next state. Since if you have multiple setState in future, you'd default to that. However, good for now.

Reply:
I am also not sure of keeping a 1 ms timer increment.

Reply:
Thanks for the feedback. The 1ms increment is grossly inaccurate. Tweaked it to 6ms(but still no way to gauge accuracy). Formatted the time to display in a better manner.